src/openrct2/rct2/S6Importer.cpp

```diff
@@ -1309,6 +1309,7 @@ class S6Importer final : public IParkImporter
         dst->colours = src->colours;
         dst->track_progress = src->track_progress;
         dst->track_direction = src->track_direction;
+        dst->BoatLocation = TileCoordsXY{ src->boat_location.x, src->boat_location.y }.ToCoordsXY();
```

Comment:
Check for null location.

Reply:
On `ride->mode != RIDE_MODE_BOAT_HIRE` this should set it to a null value.